src/scipp/core/variable.py

```diff
@@ -55,6 +55,19 @@ def scalar(value: _Any,
                          dtype=dtype)
 
 
+def index(value: _Any, dtype: _cpp.DType = None) -> _cpp.Variable:
+    """Constructs a zero dimensional :class:`Variable` representing an index.
+
+    This is equivalent to calling :py:func:`scipp.scalar` with unit=None.
+
+    :param value: Initial value.
+    :param dtype: Optional, type of underlying data. Default=None,
```

Comment:
Should we enforce `int32` or `int64`?

Reply:
I had considered that, but in the end I could not come up with a compelling argument for it, because I could not identify any risk of allowing, say, float.

I think we *do* want to support other types that default to no unit anyway. For example, someone may want to use Python tuples as indices (`sc.index((1,2))`, since `sc.scalar((1,2))` is odd), so unless we want to selectively forbid floating-point dtypes...